email_totals/handler.py

```diff
@@ -0,0 +1,315 @@
+from botocore.exceptions import ClientError
+
+# Create Cost Explorer service client using saved credentials
+ce_client = boto3.client('ce')
+
+# Create a new SES client
+ses_client = boto3.client('ses')
+
+# Name of the Cost Category containing email addresses
+email_category = 'Owner Email'
+
+# For identifying Sage users
+sagebase_email = '@sagebase.org'
+sagebio_email = '@sagebionetworks.org'
+synapse_email = '@synapse.org'
+synapse_team_sage = '273957'
+
+# Create a Synapse client
+syn_client = syn.Synapse(cache_root_dir='/tmp/synapse')
+
+#--------------------------------------------------------------------------------------------------
+# Reporting periods
+
+# This will run at the beginning of the month, looking at the previous month
+# and comparing change to the month before that.
+
+# The Start date is inclusive, and the End date is exclusive
+
+today = datetime.now()
+
+target_month = {}
+compare_month = {}
+
+# Special-case the two cases where we cross year boundaries
+if today.month == 1:
+    # in Jan, look at Dec and Nov of last year
+    target_month['Start'] = f'{today.year - 1}-12-01'
+    target_month['End'] = f'{today.year}-01-01'
+
+    compare_month['Start'] = f'{today.year - 1}-11-01'
+    compare_month['End'] = f'{today.year - 1}-12-01'
+
+elif today.month == 2:
+    # in Feb, look at Jan of this year and Dec of last year
+    target_month['Start'] = f'{today.year}-01-01'
+    target_month['End'] = f'{today.year}-02-01'
+
+    compare_month['Start'] = f'{today.year - 1}-12-01'
+    compare_month['End'] = f'{today.year}-01-01'
+
+else:
+    # no year boundary, look at the previous two months
+    target_month['Start'] = f'{today.year}-{(today.month - 2):02}-01'
+    target_month['End'] = f'{today.year}-{(today.month - 1):02}-01'
+
+    compare_month['Start'] = f'{today.year}-{(today.month - 3):02}-01'
+    compare_month['End'] = f'{today.year}-{(today.month - 2):02}-01'
+
+print(f"Target month: {target_month}")
+print(f"Compare month: {compare_month}")
+
+
+# get list of synapse users in team Sage
+
+def get_team_sage_emails():
+    '''
+    Get a list of Team Sage emails from Synapse
+    '''
+
+    team_sage = []
+
+    syn_team = syn_client.getTeam(synapse_team_sage)
+    syn_members = syn_client.getTeamMembers(syn_team)
+    for m in syn_members:
+        email = m['member']['userName'] + synapse_email
+        team_sage.append(email)
+
+    return team_sage
+
+
+# Get cost information
+
+def get_ce_costinfo(period):
+    '''
+    Get email cost category information for the given time period
+    '''
+
+    # Retrieve cost and usage metrics for specified account
+    response = ce_client.get_cost_and_usage(
+        TimePeriod=period,
+        Granularity='MONTHLY',
+        Metrics=[
+            'UnblendedCost',
+        ],
+        GroupBy=[{
+            'Type': 'COST_CATEGORY',
+            'Key': email_category,
+        }],
+    )
+
+    return response['ResultsByTime']
+
+
+def get_email_totals(costinfo):
```

Comment:
this function is getting a list of resource spend by all synapse users?  would `get_user_spending` or `get_synapse_user_spend` make more sense?

Reply:
I would expect "user spend" to include details about individual resource usage, not just a flat total. I've renamed the function `get_summary_by_email` since the dictionary will also include resources missing a `CostCenterOther` tag in addition to the user total.